option java_outer_classname = "AB";

service AddressBookService {
    rpc __error__(VoidCall) returns(ErrorReply);
    rpc AddPerson(AddPersonCall) returns(AddPersonReply);
    rpc AddPeople(AddPeopleCall) returns(AddPeopleReply);
}

message VoidCall {}

message AddPersonCall {
    required Person person = 1;
    optional string some_value = 2;
}

message AddPersonReply {
    optional int32 id = 2;
}

message ErrorReply {
    required string errorMessage = 1;
}

message Person {
    required string name = 1;
    optional string email = 2;
}

message AddPeopleCall {
	repeated Person people = 1;
	repeated string test_values = 2;
}

message AddPeopleReply {
	repeated int32 length_name = 1;
}